test-suite/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ThreadLocalRandomTest.scala

```diff
@@ -0,0 +1,119 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.util.concurrent
+
+import org.scalajs.jasminetest.JasmineTest
+import java.util.concurrent.ThreadLocalRandom
+import scala.util.Random
+import scala.math.abs
+import scala.math.max
+import scala.math.min
+
+import org.scalajs.testsuite.javalib.ExpectExceptions
+
+object ThreadLocalRandomTest extends JasmineTest with ExpectExceptions {
+
+  describe("java.util.concurrent.ThreadLocalRandom") {
+
+    it("should throw exception on setSeed") {
+      val tlr = ThreadLocalRandom.current()
+
+      expectThrows[UnsupportedOperationException](tlr.setSeed(1))
+    }
+
+    it("should return nextInt that fits bounds") {
+      val tlr = ThreadLocalRandom.current()
+
+      for (_ <- 1 to 100) {
+        val one = Random.nextInt()
+        //to have at least one positive for bound
+        val two = abs(Random.nextInt())
```

Comment:
In general, it is not a very good idea to have tests depend on random input data. I would prefer if you generated once, offline, 100 pairs of `(least, bound)`, and you hard-coded that pair in the test.

Use this opportunity to include in the pairs some known corner cases (preferably visually separated from the random ones), like `least == Int.MinValue && bound == Int.MaxValue` for example.

Same for the other tests.